### PR TITLE
Harmonize RESTMapper creation, remove DiscoveryConfiguration

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area quality
/kind cleanup
/priority normal

**What this PR does / why we need it**:
This PR removes the disk cached discovery clients in `gardener-scheduler`, `gardener-controller-manager` and `gardenlet` and makes use of the controller-runtime's dynamic RESTMapper.

**Which issue(s) this PR fixes**:
Part of #2414 

**Special notes for your reviewer**:

**Release note**:

```action operator
The `discovery` section has been removed completely from the `GardenletConfiguration`, `SchedulerConfiguration`, `ControllerManagerConfiguration` and the respective charts in `charts/gardener` as it is no longer needed. Please adapt your chart usages and/or component configurations accordingly.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Screen Shot 2020-06-05 at 08 09 02
I like such PRs :)

File: pkg/client/kubernetes/client.go [hunk 1]
```diff
@@ -316,3 +321,22 @@ func new(conf *config) (Interface, error) {
 
 	return clientSet, nil
 }
+
+func setConfigDefaults(conf *config) error {
+	if err := setClientOptionsDefaults(conf.restConfig, &conf.clientOptions); err != nil {
+		return err
+	}
+
+	return nil
```
Comment (rfranzke) [inline pkg/client/kubernetes/client.go]:
```suggestion
	return setClientOptionsDefaults(conf.restConfig, &conf.clientOptions)
```

nitpicking... :)

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]

**Review by timuthy — APPROVED** [review]
/lgtm

**Pull request merged**